src/features/layout/update/updateFormLayoutSagas.ts

```diff
@@ -279,13 +282,21 @@ export function* updateRepeatingGroupsSaga({
   }
 }
 
+const UIConfigSelector: (store: IRuntimeStore) => IUiConfig = (store: IRuntimeStore) => store.formLayout.uiConfig;
+
 export function* updateCurrentViewSaga({
   payload: { newView, runValidations, returnToView, skipPageCaching, keepScrollPos, focusComponentId },
 }: PayloadAction<IUpdateCurrentView>): SagaIterator {
   try {
-    // When triggering navigation to the next page, we need to make sure there are no unsaved changes. The action to
-    // save it should be triggered elsewhere, but we should wait until the state settles before navigating.
-    yield waitFor((state) => !state.formData.unsavedChanges);
+    const uiConfig: IUiConfig = yield select(UIConfigSelector);
+
+    if (uiConfig.autoSaveBehavior === 'onChangePage') {
+      yield call(putFormData, {});
+    } else {
```

Comment:
Have you tested this for stateless apps? The regular saveFormDataSaga uses a different saving method for stateless apps.

![image](https://github.com/Altinn/app-frontend-react/assets/47412359/5f5436c7-e13f-4b8b-b1dc-c7844cdb5912)